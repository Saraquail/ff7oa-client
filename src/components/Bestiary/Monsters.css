* {
  text-align: center;
}

#expand {
  font-size: 14px;
  width: 15em;
}

#expand:hover, #expand:focus {
  color: #26437a;
}

.monster {
  border: solid 1px;
  border-radius: 25px;
  margin: auto;
  margin-bottom: 2em;
  padding: 1em;
  width: 50%
}

