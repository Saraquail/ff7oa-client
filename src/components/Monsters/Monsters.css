* {
  text-align: center;
}

#expand {
  font-size: 14px;
  width: 13em;
  margin: auto;
}

#expand:hover, #expand:focus {
  color: #26437a;
}

.monster {
  border: solid 1px;
  border-radius: 25px;
  margin: auto;
  margin-bottom: 2em;
  padding: 1em;
  width: 50%;
  font-weight: 700;
  text-shadow: 1px -.5px #333;
}

