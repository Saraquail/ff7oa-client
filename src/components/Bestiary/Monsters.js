import React, { Component } from 'react'
import { HashLink as Link } from 'react-router-hash-link';
import SaveGuideForm from '../SaveGuideForm/SaveGuideForm'
import './Monsters.css'

class Monsters extends Component {
  state = {
    showModal: false,
    selectedid: '',
    expanded: false,
    sort: ''
  }

  handleExpand = () => {
    this.setState(prevState => ({
      expanded: !prevState.expanded
    }))
  }

  handleOpenModal = (e) => {
    let id = e.target.value
    this.setState({
      showModal: true,
      selectedid: id
    })
  }

  handleCloseModal = () => {
    this.setState({
      showModal: false
    })
  }

  renderAllMons = () => {

    const mon = this.props
    let id = this.state.selectedid
    if(!mon) {
      return <p>Loading Monsters</p>
    }

    return(
      <section className="monsterList" >
        <div className="monster">
          <div className="collapsed">    
            <h2 id={mon.id}>{mon.name}</h2>
            <button value={mon.id}
              onClick={e => this.handleOpenModal(e)}
              > Add to My PHS</button>
            <Link to={'#' + mon.id} id="expand">
              <button id="expand" onClick={this.handleExpand}>
              Click for more/less info
              </button>
            </Link> 
            <p> Level: {mon.level} </p>
            <p> Added By: {mon.user_name} </p>
            <p> HP: {mon.hp} MP: {mon.mp} </p>
            <p> Exp: {mon.exp} Gil: {mon.gil} </p>
            <p> Strength: {mon.strength}</p>
            <p> Weakness: {mon.weakness}</p>
          </div>
          {this.state.expanded 
            ? <div className="expanded">
              <p> Location: {mon.location} </p>
              <p> Steal: {mon.steal} </p>
              <p> Drop: {mon.drops} </p>
              <p> Enemy_skill: {mon.enemy_skill} </p>
            </div>
            : ''}
        </div>
      {this.state.showModal ?
        <SaveGuideForm className="modal overlay"  selectedid={id} handleCloseModal={this.handleCloseModal}   onRequestClose={this.handleCloseModal}
        />
      : ''}
      </section>

    )
  }


  render () {

    return (
      <div>
      
      {this.renderAllMons()}
      </div>
    )
    

  }
}

export default Monsters