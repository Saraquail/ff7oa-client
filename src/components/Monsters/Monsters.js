import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { HashLink as Link } from 'react-router-hash-link';
import SaveGuideForm from '../SaveGuideForm/SaveGuideForm';
import './Monsters.css';

class Monsters extends Component {
  state = {
    showModal: false,
    selectedid: '',
    expanded: false,
    // sort: '',
  }

  handleExpand = () => {
    this.setState((prevState) => ({
      expanded: !prevState.expanded,
    }));
  }

  handleOpenModal = (e) => {
    const id = e.target.value;
    console.log(e.target.value);
    this.setState({
      showModal: true,
      selectedid: id,
    });
  }

  handleCloseModal = () => {
    this.setState({
      showModal: false,
    });
  }

  renderAllMons = () => {
    const {
      id,
      name,
      hp,
      mp,
      exp,
      ap,
      gil,
      weakness,
      strength,
      location,
      level,
      drops,
      steal,
      morph,
      enemy_skill,
    } = this.props;
    if (!name) {
      return <p>Loading Monsters</p>;
    }

    const { expanded, showModal } = this.state;

    return (
      <section className="monsterList">
        <div className="monster">
          <div className="collapsed">
            <h2 id={id}>{name}</h2>
            <button
              type="button"
              value={id}
              onClick={(e) => this.handleOpenModal(e)}
            >
              {' '}
              Add to My PHS
            </button>
            <Link to={`#${id}`} id="expand">
              <button type="button" id="expand" onClick={this.handleExpand}>
                Click for more/less info
              </button>
            </Link>
            {' '}
            <p>
              {' '}
              Level:
              {level}
            </p>
            <p>
              {' '}
              HP:
              {hp}
              {' '}
              MP:
              {mp}
              {' '}
            </p>
            <p>
              {' '}
              Exp:
              {exp}
              {' '}
              Ap:
              {ap}
            </p>
            <p>
              Gil:
              {gil}
            </p>
            <p>
              {' '}
              Strength:
              {strength}
            </p>
            <p>
              {' '}
              Weakness:
              {weakness}
            </p>
          </div>
          {expanded
            ? (
              <div className="expanded">
                <p>
                  {' '}
                  Location:
                  {location}
                </p>
                <p>
                  {' '}
                  Steal:
                  {steal}
                </p>
                <p>
                  {' '}
                  Drop:
                  {drops}
                  {' '}
                  Morph:
                  {morph}
                </p>
                <p>
                  {' '}
                  Enemy_skill:
                  {enemy_skill}
                </p>
              </div>
            )
            : ''}
        </div>
        {showModal
          ? (
            <SaveGuideForm
              className="modal overlay"
              selectedid={id}
              handleCloseModal={this.handleCloseModal}
              onRequestClose={this.handleCloseModal}
            />
          )
          : ''}
      </section>

    );
  }


  render() {
    return (
      <div>
        {this.renderAllMons()}
      </div>
    );
  }
}

Monsters.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  hp: PropTypes.number,
  mp: PropTypes.number,
  exp: PropTypes.number,
  ap: PropTypes.number,
  gil: PropTypes.number,
  weakness: PropTypes.string,
  strength: PropTypes.string,
  location: PropTypes.string,
  level: PropTypes.number,
  drops: PropTypes.string,
  steal: PropTypes.string,
  morph: PropTypes.string,
  enemy_skill: PropTypes.string,
};

export default Monsters;
