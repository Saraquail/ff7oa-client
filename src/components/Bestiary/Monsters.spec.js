import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Monsters from './Monsters.js';

describe('<Monsters/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <Monsters />
      </BrowserRouter>, div,
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Monsters />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
