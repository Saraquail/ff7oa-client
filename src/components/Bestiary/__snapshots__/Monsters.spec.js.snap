// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Monsters/> renders the UI as expected 1`] = `
<div>
  <section
    className="monsterList"
  >
    <div
      className="monster"
    >
      <div
        className="collapsed"
      >
        <h2 />
        <button
          onClick={[Function]}
        >
           Add to My PHS
        </button>
        <a
          href="/#undefined"
          id="expand"
          onClick={[Function]}
        >
          <button
            id="expand"
            onClick={[Function]}
          >
            Click for more/less info
          </button>
        </a>
         
        <br />
        <p>
           Level: 
           
        </p>
        <p>
           Added By: 
           
        </p>
        <p>
           HP: 
           MP: 
           
        </p>
        <p>
           Exp: 
           Gil: 
           
        </p>
        <p>
           Strength: 
        </p>
        <p>
           Weakness: 
        </p>
      </div>
      
    </div>
    
  </section>
</div>
`;
